### PR TITLE
Cities autocomplete client

### PR DESCRIPTION
**Pull request by adrianburcin**
Added a new component named Dashboard which will be the default route
Input debounce component which will emit an event if the input was changed and after a delay specified
Cities autocomplete service which will call the ../api/cities/<word> endpoint to create a draft autocomplete like items
Remove typescript uglish types from constructor and add static parameters method to inject services into constructor
Provide a apiUrl object from main.js using provide service as value to be used in all components by injecting 'apiUrls' in constructor

### CODE REVIEW

File: src/app/main.js [hunk 1]
```diff
@@ -3,7 +3,9 @@ import { bootstrap } from 'angular2/platform/browser';
 import { ROUTER_PROVIDERS, LocationStrategy, HashLocationStrategy } from 'angular2/router';
 
 import { AppComponent } from './app.component';
+import { apiUrls } from './configs/api';
 
 bootstrap(AppComponent, [ROUTER_PROVIDERS,
-  provide(LocationStrategy, { useClass: HashLocationStrategy })
+  provide(LocationStrategy, { useClass: HashLocationStrategy }),
+  provide('apiUrls', { useValue: apiUrls })
```
Comment (monostere0) [inline src/app/main.js]: You should provide this dependancy only to the DAL services.

File: src/app/heroes/hero-list.component.js [hunk 2]
```diff
@@ -14,15 +16,19 @@ import { HeroService } from './hero.service';
...
 export class HeroListComponent implements OnInit {
   heroes;
   selectedId;
 
-  constructor(service:HeroService,
-              router:Router,
-              routeParams:RouteParams) {
+  // Angular 2 DI
+  static get parameters() {
+    return [HeroService, Router, RouteParams, 'apiUrls'];
```
Comment (monostere0) [inline src/app/heroes/hero-list.component.js]: This component makes no use of 'apiUrls'.

File: src/app/main.js (hunk 1 above)
Comment (adrianburcin) [inline src/app/main.js]: Fixed. Moved to cities.service as import

File: src/app/heroes/hero-list.component.js (hunk 2 above)
Comment (adrianburcin) [inline src/app/heroes/hero-list.component.js]: Fixed

**Comment by monostere0** [comment]
Dashboard, Crisis, Heroes stuff, they look very strange (and totally unrelated) to me. We should just keep what's related to the weather and get rid of the rest, do you agree?
Maybe move the heroes stuff on a separate branch and leave the master clean, without such mixes.

**Comment by adrianburcin** [comment]
Unrelated components were removed and readme file was updated :)

**Comment by monostere0** [comment]
> Unrelated components were removed and readme file was updated :)

Great dude!

**Comment by monostere0** [comment]
LGTM! 👍

**Pull request merged**